ncm-systemd/src/main/pan/components/systemd/schema.pan

```diff
@@ -233,6 +247,77 @@ type ${project.artifactId}_unitfile_config_service = {
     true;
 };
 
+@documentation{
+the [Socket] section
+http://www.freedesktop.org/software/systemd/man/systemd.socket.html
+}
+type ${project.artifactId}_unitfile_config_socket = {
+    include ${project.artifactId}_unitfile_config_systemd_exec
+    include ${project.artifactId}_unitfile_config_systemd_kill
+    'ListenStream' ? string[]
+    'ListenDatagram' ? string[]
+    'ListenSequentialPacket' ? string[]
+    'ListenFIFO' ? string with match(SELF, '^/')
+    'ListenSpecial' ? string with match(SELF, '^/')
+    'ListenNetlink' ? string
+    'ListenMessageQueue' ? string with match(SELF, '^/')
+    'ListenUSBFunction' ? string
+    'SocketProtocol' ? string with match(SELF, '^(udplite|sctp)$')
+    'BindIPv6Only' ? string with match(SELF, '^(default|both|ipv6-only)$')
+    'Backlog' ? long(0..)
+    'BindToDevice' ? string
+    'SocketUser' ? defined_user
+    'SocketGroup' ? defined_group
+    'SocketMode' ? long # octal
```

Comment:
these probably can also be limited `(0..1023)` (ot sure anyone wants to set the sticky bit, otherwise stop at 511